common4j/src/main/com/microsoft/identity/common/java/flighting/CommonFlight.java

```diff
@@ -88,7 +88,12 @@ public enum CommonFlight implements IFlightConfig {
     /**
      * Flight to enable the re-attachment of new PRT header logic. Default is true.
      */
-    ENABLE_ATTACH_NEW_PRT_HEADER_WHEN_NONCE_EXPIRED("EnableAttachNewPrtHeaderWhenNonceExpired", true);
+    ENABLE_ATTACH_NEW_PRT_HEADER_WHEN_NONCE_EXPIRED("EnableAttachNewPrtHeaderWhenNonceExpired", true),
+
+    /**
+     * Flight to enable the new key generation spec for wrap key. Default is true.
+     */
+    ENABLE_NEW_KEY_GEN_SPEC_FOR_WRAP("EnableNewKeyGenSpecForWrap", true);
```

Comment:
I actually think we need to also have a flight for the new change here (whether to use purpose for new key spec).

So a total of 2 flights:

- 1 for enabling new key spec
- 1 for setting purpose wrap or not

This is because new key spec is already on 25% devices with purpose wrap, right? Are we okay to move forward with this change that says either turn on new spec without wrap purpose or just turn off new spec? Why not also have the option to have wrap purpose just in case the new spec without wrap purpose causes problems?

Reply:
As we discussed offline, I am going to skip using 2 flights and rely on try catch blocks and just have 1 flight.